packages/block-library/src/paragraph/edit.js

```diff
@@ -145,10 +123,58 @@ function ParagraphBlock( {
 						  )
 				}
 				data-empty={ content ? false : true }
-				placeholder={ placeholder || __( 'Type / to choose a block' ) }
-				__unstableEmbedURLOnPaste
-				__unstableAllowPrefixTransformations
-			/>
+			>
+				{ ! content && (
+					<DropZone
```

Comment:
The `<DropZone>` component is put inside the `<p>` tag to position itself. I'm not sure if this approach will impact the block's HTML tree though. An alternative is to use `<Popover>` to render the `<DropZone>` component instead.

Reply:
Hmm, yeah, not sure about the HTML changes here. Quite a lot of effort was put into making the paragraph block just a `<p>`

Could an option be to use `__experimentalUseDropZone` to avoid those changes?

I guess that means reimplementing quite a lot of the visual aspects of drop zone, unless that can be broken down into smaller parts. 🤔

Reply:
Yeah, I just found out that changing the structure of the block breaks a bunch of tests. 😞

I'll see what I can do 🤔